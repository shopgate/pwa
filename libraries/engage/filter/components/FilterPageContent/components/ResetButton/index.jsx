import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { I18n, Button, SurroundPortals } from '@shopgate/engage/components';
import { PORTAL_FILTER_RESET_BUTTON } from '@shopgate/engage/filter/constants';
import styles from './style';

/**
 * The filter reset button component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const FilterResetButton = ({ active, onClick }) => (
  <SurroundPortals
    portalName={PORTAL_FILTER_RESET_BUTTON}
    portalProps={{ active, onClick }}
  >
    <div className={styles}>
      <Button
        flat
        type="primary"
        onClick={onClick}
        disabled={!active}
        testId="clearAllButton"
      >
        <I18n.Text string="filter.reset" />
      </Button>
    </div>
  </SurroundPortals>
);

FilterResetButton.propTypes = {
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default memo(FilterResetButton);
