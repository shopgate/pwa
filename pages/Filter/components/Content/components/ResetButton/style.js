import { css } from 'glamor';

export default css({
  marginTop: 4,
  marginBottom: 40,
  textAlign: 'right',
});
