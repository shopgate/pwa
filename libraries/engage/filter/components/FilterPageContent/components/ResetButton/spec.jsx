import React from 'react';
import { shallow } from 'enzyme';
import ResetButton from './index';

const clickMock = jest.fn();

jest.mock('@shopgate/engage/components');

describe('Filter: <ResetButton />', () => {
  it('should render as activated', () => {
    const wrapper = shallow(<ResetButton active onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as deactivated', () => {
    const wrapper = shallow(<ResetButton active={false} onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    const wrapper = shallow(<ResetButton onClick={clickMock} active />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Button').simulate('click');
    expect(clickMock).toHaveBeenCalled();
  });
});
