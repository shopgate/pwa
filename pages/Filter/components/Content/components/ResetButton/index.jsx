import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { I18n, Button } from '@shopgate/engage/components';
import styles from './style';

/**
 * The filter reset button component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const FilterResetButton = ({ active, onClick }) => (
  <div className={styles}>
    <Button
      flat
      type="primary"
      onClick={onClick}
      disabled={!active}
      testId="clearAllButton"
    >
      <I18n.Text string="filter.reset" />
    </Button>
  </div>
);

FilterResetButton.propTypes = {
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default memo(FilterResetButton);
