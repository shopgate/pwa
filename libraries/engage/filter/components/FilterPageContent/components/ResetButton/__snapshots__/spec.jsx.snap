// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter: <ResetButton /> should handle clicks 1`] = `
<SurroundPortals
  portalName="filter.reset-button"
  portalProps={
    Object {
      "active": true,
      "onClick": [MockFunction],
    }
  }
>
  <div
    className={
      Object {
        "data-css-lg74ao": "",
      }
    }
  >
    <Button
      disabled={false}
      flat={true}
      onClick={[MockFunction]}
      testId="clearAllButton"
      type="primary"
    >
      <Text
        string="filter.reset"
      />
    </Button>
  </div>
</SurroundPortals>
`;

exports[`Filter: <ResetButton /> should render as activated 1`] = `
<SurroundPortals
  portalName="filter.reset-button"
  portalProps={
    Object {
      "active": true,
      "onClick": [Function],
    }
  }
>
  <div
    className={
      Object {
        "data-css-lg74ao": "",
      }
    }
  >
    <Button
      disabled={false}
      flat={true}
      onClick={[Function]}
      testId="clearAllButton"
      type="primary"
    >
      <Text
        string="filter.reset"
      />
    </Button>
  </div>
</SurroundPortals>
`;

exports[`Filter: <ResetButton /> should render as deactivated 1`] = `
<SurroundPortals
  portalName="filter.reset-button"
  portalProps={
    Object {
      "active": false,
      "onClick": [Function],
    }
  }
>
  <div
    className={
      Object {
        "data-css-lg74ao": "",
      }
    }
  >
    <Button
      disabled={true}
      flat={true}
      onClick={[Function]}
      testId="clearAllButton"
      type="primary"
    >
      <Text
        string="filter.reset"
      />
    </Button>
  </div>
</SurroundPortals>
`;
